components/datetime/src/provider/mod.rs

```diff
@@ -28,6 +29,10 @@ pub mod key {
     pub const GREGORY_DATE_PATTERNS_V1: ResourceKey =
         resource_key!(DateTime, "gregory_patterns", 1);
```

Comment:
Suggestion: I have a separate `thought` about this below, but I'm promoting it to a suggestion here on my second pass through the code. `"gregory_patterns"` is ambiguous here, and I would prefer something more precise like `"gregory_length_patterns"`. I was finding myself confused about the use of "patterns" as a term for the data payloads, as the skeleton information _also_ point to patterns.

Reply:
renamed to `gregory_lengths`. The `resource_key` is limited to TinyStr16